src/main/java/io/codelex/dateandtimepractice/DatePeriod.java

```diff
@@ -0,0 +1,73 @@
+package io.codelex.dateandtimepractice;
+
+import java.time.LocalDate;
+
+public class DatePeriod {
+
+    private LocalDate start;
+    private LocalDate end;
+
+    public DatePeriod(LocalDate start, LocalDate end) {
+        this.start = start;
+        this.end = end;
+    }
+
+    public LocalDate getStart() {
+        return start;
+    }
+
+    public LocalDate getEnd() {
+        return end;
+    }
+
+    @Override
+    public String toString() {
+
+
+        return "overlap start date " + String.format("%02d",start.getDayOfMonth()) + "." + String.format("%02d",start.getMonthValue()) + "." + start.getYear()
+                + " and end date " + String.format("%02d",end.getDayOfMonth()) + "." + String.format("%02d",end.getMonthValue()) + "." + end.getYear();
+    }
+
+    public DatePeriod intersection(DatePeriod period) {
+        if (this.end.isBefore(period.getStart()) || period.getEnd().isBefore(this.start)) {
+            return null;
+        } else if (this.end.isEqual(period.getStart())) {
+            return new DatePeriod(this.end, period.getStart());
+        } else if (this.start.isEqual(period.getEnd())) {
+            return new DatePeriod(this.start, period.getEnd());
+        } else if (this.start.isBefore(period.getStart()) && this.end.isAfter(period.getEnd())) {
+            return new DatePeriod(period.getStart(), period.getEnd());
+        } else if (this.start.isAfter(period.getStart()) && this.end.isBefore(period.getEnd())) {
+            return new DatePeriod(this.start, this.end);
+        } else {
+            LocalDate startOverlap = null;
```

Comment:
bez šī var iztikt un if blokā
`return new DatePeriod(period.getStarted(), this.end)`

Reply:
Īsti neiebraucu tavu domu... Līdzīgi kā notOverLapping() šo gadījumu pārnesu uz boolean palīgmetodi.

Reply:
domāju šādi, un arī īsti nevajdzētu modificēt un atgriezt topašu objektu šajā gadijumā, jo idejiski šis ir jauns overalppin periods, bet padot ķip 2. periodu
```
if (period.getStart().isBefore(this.end)) {
   return new DatePeriod(period.getStart(), this.end)
} else if (period.getEnd().isAfter(this.start)) {
   return new DatePeriod(this.start, period.getEnd())
}